packages/sdk-viem/src/actions.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  EthBridger,
+  ParentToChildMessageStatus,
+  ParentTransactionReceipt,
+} from '@arbitrum/sdk'
+import {
+  publicClientToProvider,
+  viemTransactionReceiptToEthersTransactionReceipt,
+} from '@offchainlabs/ethers-viem-compat'
+import { BigNumber } from 'ethers'
+import {
+  Account,
+  Address,
+  Hash,
+  PublicClient,
+  TransactionRequest,
+  WalletClient,
+} from 'viem'
+
+export type PrepareDepositEthParameters = {
+  amount: bigint
+  account: Account | Address
+}
+
+const DEFAULT_CONFIRMATIONS = 1
+const DEFAULT_TIMEOUT = 1000 * 60 * 5 // 5 minutes
+
+export type WaitForCrossChainTxParameters = {
+  hash: Hash
+  timeout?: number
+  confirmations?: number
+}
+
+export type SendCrossChainTransactionParameters = {
+  request: TransactionRequest
+  timeout?: number
+  confirmations?: number
+}
+
+export type CrossChainTransactionStatus = {
+  status: 'success' | 'failed'
+  complete: boolean
+  message?: unknown
+  childTxReceipt?: unknown
+  hash: Hash
+}
+
+export type DepositEthParameters = {
+  amount: bigint
+  account: Account | Address
+  confirmations?: number
+  timeout?: number
+}
+
+export type ArbitrumDepositActions = {
+  prepareDepositEthTransaction: (
+    params: PrepareDepositEthParameters
+  ) => Promise<TransactionRequest>
+}
+
+export type ArbitrumParentWalletActions = {
+  waitForCrossChainTransaction: (
+    params: WaitForCrossChainTxParameters
+  ) => Promise<CrossChainTransactionStatus>
+
+  sendCrossChainTransaction: (
+    params: SendCrossChainTransactionParameters
+  ) => Promise<CrossChainTransactionStatus>
+
+  depositEth: (
+    params: DepositEthParameters
+  ) => Promise<CrossChainTransactionStatus>
+}
+
+async function prepareDepositEthTransaction(
```

Comment:
I would still reexport the actions on their own. It helps with treeshaking and it's common viem API to have both decorators and actions, `https://viem.sh/docs/clients/custom#tree-shaking`